### PR TITLE
Update to current docs using automated tool

### PR DESCRIPTION
**Pull request by DannyvdSluijs**
### This WIP PR is fully generated using a new tool I've been working one https://github.com/DannyvdSluijs/PicqerExactPhpClientGenerator

This PR:
- Updates the property documentation
- Adds the missing properties
- Removes the obsolete properties
- Removes the `Persistance\Storable` trait on models that only support the `GET` method
- Updates the ordering of the code (traits, properties, functions)
- Updates property type from `binary` to `string` as binary isn't a PHP type.
- Updates property type from `int64` to `int` as int64 isn't a PHP type.
- Updates the collection types from a non-existing class e.g. `BankEntryLines` to `BankEntryLine[]`
- Updates the order of the `fillable` properties to match the documentation order.
- Adds missing models (`CustomerItem`, `DeductibilityPercentage`, `DigitalOrderPicking`, `DigitalOrderPickingLine`, `EmploymentEndReasonsOnFocusDate`, `SalesChannel`, `SystemUser`)
- Removed identical overloads of the `primaryKey` property.

Some items that still need work
- [x] Removing code to support downloadable items
- [x] Removing code to set collection items ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-46c3419ff032ca8dc3d3310ecc6e53e33bee2a652c4e13a06f945a851ff96425L59) and [here](https://github.com/picqer/exact-php-client/pull/489/files#diff-25dab0c6e202bb1c4a073883daca7ac8e68fe9f817c53f49bb456519b5fd173fL52))
- [x] Support hyperlinks in property description in the api docs ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-ae7f6cf1d406682debb8dbeb2845b8b84e04c8a3c3d144f85ee42905d07b1a33R11))
- [x] Keep obsolete properties (if needed ???) See https://github.com/picqer/exact-php-client/pull/489#issuecomment-888293876, obsolete properties can be removed.
- [x] Avoid actual comments being removed ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-59c470529ad7c6b64e82fca88693b982d481b414173be198dd53f6e0bc80a8b5L78))
- [x] Add test cases for the missing models that where added
- [x] Find out a solution as `src/Picqer/Financials/Exact/Division.php` is replaced with `SystemSystemDivisions` due to naming convention "mismatch" Same goes for `UserUsers` and `SystemUsers`.
- [x] Resolve case where incorrect class names are suggested ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-cf6655e07a949e7799235d2eeec08af8790715c23e33b7d9552aa83035dd0dd1R11))
- [x] Avoid the `Me` model to have its functions removed.
- [x] Avoid save and insert overload being removed. ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-64f4271419018c94be8d34b5f2d7bb98a4c376f7548b5e2cb77d67be8f8d1896L61), [here](src/Picqer/Financials/Exact/PrintedSalesOrder.php))
- [x] Deal with models that have a compound key ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-d55eb62007da00a5ffe999fa4f17ed88746440021350d2b6158e4189d6c23b00L19))
- [x] Avoid `@deprecated` tags to be removed ([here](https://github.com/picqer/exact-php-client/pull/489/files#diff-43341dc12d693c213d6799153319c2b99bd6b985080ff75e5c69623b278e1a0eL9))
- [x] Avoid newlines being tossed when "copying" code from current file to new content

### CODE REVIEW

File: src/Picqer/Financials/Exact/BankEntry.php [hunk 1]
```diff
@@ -33,8 +33,6 @@ class BankEntry extends Model
 
     protected $primaryKey = 'EntryID';
 
-    protected $bankEntryLines = [];
```
Comment (DannyvdSluijs) [inline src/Picqer/Financials/Exact/BankEntry.php:36]: This should not be deleted. Perhaps for all collections properties such an internal property is needed although this property is unused (never initialised, never accessed).

**Comment by DannyvdSluijs** [comment]
To avoid any code being removed the AST parser written by Nikita Popov could help. See comment https://github.com/nikic/PHP-Parser/issues/110#issuecomment-41032348

File: src/Picqer/Financials/Exact/SystemUser.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        'PhoneExtension',
+        'ProfileCode',
+        'StartDate',
+        'Title',
+        'UserDivisionList',
+        'UserName',
+        'UserTypeCode',
+    ];
+
+    protected $url = 's';
```
Comment (DannyvdSluijs) [inline src/Picqer/Financials/Exact/SystemUser.php:93]: Find out why url is wrong.

**Comment by DannyvdSluijs** [comment]
@stephangroen I would like your opinion on this PR. As I'm going towards completing the PR checklist 
some input would be appreciated.

If needed the tool I've build to automate these updates allows for single endpoint update. This means it can be 143 individual PR to keep the changes small.

**Comment by stephangroen** [comment]
@DannyvdSluijs Nice tool! I think it's great to instantly have an up to date client according to the Exact docs. Don't see any objections to merging this when you're done.

**Comment by stephangroen** [comment]
Of the 2 remaining todo's, adding tests would be great. Although they don't test that much, it at least makes sure there are no code errors and the class can be instantiated. 

Keeping the obsolete properties, not entirely sure. They are already obsolete and cannot be used anymore and since the getters and setters are magic, it probably won't change anything if someone still uses them. They should already be empty and not cause an error. I think we can remove them.

**Comment by DannyvdSluijs** [comment]
I'm very happy report this PR is now complete. Hoping for a thumbs up review and merge.

In order to re-update the repository in the future on can run my tool again and review the changes to code before creating a PR. See https://github.com/DannyvdSluijs/PicqerExactPhpClientGenerator for the tool I've build. Not the quality I would like it to have but it does the job. You can run the tool using `./generate ../ExactOnline/exact-php-client --refresh-meta-data`

**Comment by remkobrenters** [comment]
+1 Great work. Thank you!

**Comment by DannyvdSluijs** [comment]
Anything you need from me before this can be merged?

**Pull request merged**